driver-legacy/src/main/com/mongodb/Mongo.java

```diff
@@ -324,7 +326,9 @@ public Mongo(
         this.credentialsList = unmodifiableList(credentialsList);
 
         AutoEncryptionSettings autoEncryptionSettings = options.getAutoEncryptionSettings();
-        this.delegate = new MongoClientDelegate(cluster, credentialsList, this,
+        this.delegate = new MongoClientDelegate(cluster,
+                createRegistry(options.getCodecRegistry(), options.getUuidRepresentation()),
```

Comment:
Should this be moved to the options.builder().build() stage?

Reply:
We could go either way on this.  I didn't do it for a couple of reasons:

1. It matches what we have to do in MongoCollectionImpl/MongoDatabaseImpl#withCodecRegistry
2. It preserves the application-provided CodecRegistry as-is, so if for whatever reason an application wanted to grab the registry from a built MongoClientSettings/MongoClientOptions, it wouldn't carry along the UuidRepresentation with it when you call MongoClientOptions/Settings#getCodecRegistry.